drivers/flash/flash_shell.c

```diff
@@ -220,10 +267,12 @@ static int cmd_test(const struct shell *shell, size_t argc, char *argv[])
 }
 
 SHELL_STATIC_SUBCMD_SET_CREATE(flash_cmds,
-	SHELL_CMD(erase, NULL, "<page address> <size>", cmd_erase),
-	SHELL_CMD(read, NULL, "<address> <Dword count>", cmd_read),
-	SHELL_CMD(test, NULL, "<address> <size> <repeat count>", cmd_test),
-	SHELL_CMD(write, NULL, "<address> <dword> <dword>...", cmd_write),
+	SHELL_CMD(erase, NULL, "[<device>] <page address> [<size>]", cmd_erase),
```

Comment:
use `SHELL_CMD_ARG` to validated number of arguments. last 2 arguments are for mandatory number of commands and optional.

Reply:
ok, I pushed an updated version that uses `SHELL_CMD_ARG` to validate arg count.